server/src/main/resources/rules/rules.js

```diff
@@ -2492,6 +2573,42 @@ var Autobind = {
                     }
                 }
                 return provided;
+            },
+
+            get_addons: function() {
+                return this.get_addons_pools(this.pools);
+            },
+
+            // Returns list of all addons from the given array of pools.
+            get_addons_pools: function(in_pools) {
+                var addons = [];
+                for (var i = 0; i < in_pools.length; i++) {
+                    var pool = in_pools[i];
+                    var pool_addons = pool.retrievePoolAttributeValues("addons");
+                    for (var k = 0; k < pool_addons.length; k++) {
+                        addons.push(pool_addons[k]);
+                    }
+                }
+                log.debug("get_addons_pools:  " + addons);
+                return addons;
```

Comment:
This list could possibly have duplicates in it.  Is that something to worry about?